src/routes/jira/jira-get-connections-backfilStatus.ts

```diff
@@ -0,0 +1,128 @@
+import { NextFunction, Request, Response } from "express";
+import { groupBy, countBy } from "lodash";
+import { RepoSyncState } from "~/src/models/reposyncstate";
+// import { Subscription, TaskStatus } from "~/src/models/subscription";
+import { TaskStatus } from "~/src/models/subscription";
+
+type subscriptionBackfillState = {
+	totalRepos?: number;
+	syncCompleted?: boolean;
+	syncedRepos?: number;
+	backfillSince?: string | null;
+};
+type backFillType = {
+	[key: string]: subscriptionBackfillState;
+};
+
+export const JiraGetConnectionsBackfilStatus = async (
+	req: Request,
+	res: Response,
+	next: NextFunction
+): Promise<void> => {
+	try {
+		const subscriptionIds: number[] = String(req.query.subscriptionIds)
+			.split(",")
+			.map((id) => Number(id))
+			.filter((n) => !!n);
+
+		if (subscriptionIds.length <= 0) {
+			req.log.error("Missing Subscription IDs");
+			res.status(401).send("Missing Subscription IDs");
+			return;
+		}
+
+		// const subscriptionStatus = await Subscription.findAll({
+		// 	where: {
+		// 		id: subscriptionIds,
+		// 	}
+		// });
+		// const subscriptions = groupBy(subscriptionStatus, "id") as unknown as { fulfilled: PromiseFulfilledResult<Subscription>[], rejected: PromiseRejectedResult[] };
+
+		// if (!subscriptions) {
+		// 	req.log.error("Missing Subscription");
+		// 	res.status(401).send("Missing Subscription");
+		// 	return;
+		// }
+
+		const repoSyncStates = await RepoSyncState.findAll({
+			where: {
+				subscriptionId: subscriptionIds
+			}
+		});
+
+		const connections = groupBy(
+			repoSyncStates,
+			"subscriptionId"
+		) as unknown as {
+			fulfilled: PromiseFulfilledResult<RepoSyncState>[];
+			rejected: PromiseRejectedResult[];
+		};
+		const backfillStatus: backFillType = {};
+
+		for (const subscriptionId in connections) {
```

Comment:
Better to move this logic in a separate method.

Reply:
Done